src/interpreter.js

```diff
@@ -264,37 +243,37 @@ const compileArguments = (args) => args.map(compileExpression).join()
 
 const compileSentenceSequence = (sentences) => {
   const compiledSentences = sentences.map(compileSentence)
-  if(compiledSentences.length && !compiledSentences[compiledSentences.length - 1].startsWith('return'))
-    compiledSentences[compiledSentences.length - 1] = 'return ' + compiledSentences[compiledSentences.length - 1]
+  if (compiledSentences.length && !compiledSentences[compiledSentences.length - 1].startsWith('return'))
+    compiledSentences[compiledSentences.length - 1] = `return ${compiledSentences[compiledSentences.length - 1]}`
   return compiledSentences.join('')
 }
 
+const findByType = (content, type) => content.filter(c => c.nodeType === type)
+
 //TODO: Imports
-const compileFile = ({content}) => {
-  const imports = content.filter(c => c.nodeType === 'Import')
-  const programs = content.filter(c => c.nodeType === 'Program')
-  const tests = content.filter(c => c.nodeType === 'Test')
-  const elements = content.filter(c => c.nodeType !== 'Test' && c.nodeType !== 'Program' && c.nodeType != 'Import')
-
-  if(!programs.length && !tests.length) {
-    log.error(`There is no program or tests to run!`)
+const compileFile = ({ content }) => {
+  const imports = findByType(content, 'Import')
+  const programs = findByType(content, 'Program')
+  const tests = findByType('Test')
```

Comment:
I meant polymorphism from a structural perspective.
For example, from the ObjectNode point of view, all memberNodes are to be treated the same: Whatever they shape is just compile those and add them to the body. It does not have to cherry pick them or classify them, just do the same to all. Same happens with Expression nodes, you don't need to know WHICH expression you have, just compile it and send it away.

I think this case is different because you have very specific things you want to do with these three cases, so we could just store those in separate collections instead of filtering a single collection that will never be used whole.